HTML Reporter: Make it more obvious why diff is suppressed

shivamdixit opened the pull request:
If the depth of an object is more than 'Max Depth' specified in `Qunit.dump` then the diff is not displayed. This behavior is confusing. This PR will display a message to the user so that user knows why it is not displayed.

Refer: https://github.com/jquery/qunit/issues/793#issuecomment-84495125

I'm confused with what text to display to the user since the user might not be aware of max depth. Any suggestions?

Screenshot:
![qunit4](https://cloud.githubusercontent.com/assets/3061095/6769225/271d3fc6-d0b8-11e4-9bf2-c135690cbfac.png)

scottgonzalez commented:
Ideally it says what the current max depth is and provides a link to rerun with a higher max depth or no max depth.

shivamdixit commented:
@scottgonzalez I've updated the PR. Screenshot:

![qunit5](https://cloud.githubusercontent.com/assets/3061095/6795895/5f2af118-d20e-11e4-94ae-0f1a261137f5.png)

1. Added `maxDepth` in url parameter, `maxDepth = -1` implies run the tests without any `maxDepth`.

2. Although `maxDepth = someVal` in URL parameters will work, but not *explicitly* showing the option of inputting the value from user because the UI will be cluttered as we will need to add input box, button etc.

3. Instead showing `QUnit.dump.maxDepth` as hint, in case user wants to change `maxDepth`.

4. The color of `Rerun` looks dull in the message, any suggestions for a different color?

5. As `QUnit.dump` is not in scope when we are parsing URL params, hence I've added extra property `QUnit.config.maxDepth`.

@jzaefferer Please provide your valuable suggestions as well.

shivamdixit commented:
I think the tests are failing because of the timeout. How can I re-run the tests?

jzaefferer commented on reporter/html.js:
```diff
@@ -663,6 +663,15 @@ QUnit.log(function( details ) {
+				message += "<tr class='test-message'><th>Message: </th><td>" +
+					"Diff suppressed as the depth of object is more than current max depth (" +
+					QUnit.config.maxDepth + ").<p>Hint: Use 'QUnit.dump.maxDepth' to run " +
+					" with a higher max depth or <a href='?maxDepth=-1'>" +
```
This should use `setUrl()` to generate QUnit specific URLs that include other URL parameters already present (filters etc.).

jzaefferer commented:
@shivamdixit thanks for the PR. I have very little bandwidth to review anything this week. I added one comment above. I'd like to discuss the general concept for tackling this issue, but that has to wait a few days.

scottgonzalez commented:
That screenshot looks great. You should wrap `QUnit.config.maxDepth` in `<code>` instead of quotes.

Fixing the styling of the rerun link should be very simple, but let's wait for the team to discuss.

shivamdixit commented:
@jzaefferer @scottgonzalez Thanks for your suggestions. I've updated the code.

jzaefferer commented:
Let's add some CSS to display the Rerun link properly, just change `color` to something with more contrast.

shivamdixit commented:
![qunit5](https://cloud.githubusercontent.com/assets/3061095/6850203/bda8811a-d3fd-11e4-8b9e-f65076f5e97b.png)

How is it looking?

leobalter commented:
LGTM
